### PR TITLE
feat(anthos.dashboard.ips): add order configurator

### PR DESCRIPTION
**Pull request by y4nnL**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/anthos-m4`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-8316
| License          | BSD 3-Clause

- [X] Try to keep pull requests small so they can be easily reviewed.
- [X] Commits are signed-off
- [X] Only FR translations have been updated
- [X] Branch is up-to-date with target branch
- [X] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [X] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Order additional blocks of anthos public ips

### CODE REVIEW

**Comment by antleblanc** [comment]
Commit message ([<short summary>](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-header)) is not compliant with the guideline. Could you please rename it?
Thank you!

**Comment by y4nnL** [comment]
@antleblanc my bad
It's done :+1: (both commit message and PR's title)

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/template.html [hunk 1]
```diff
@@ -1,11 +1,66 @@
 <oui-modal
-    data-heading="{{:: 'anthos_tenant_dashboard_component_order_public_ip_modal_title' | translate }}"
-    data-secondary-action="$ctrl.onOrderPublicCancel()"
-    data-secondary-label="{{:: 'anthos_tenant_dashboard_component_order_public_ip_modal_action_cancel' | translate }}"
+    class="anthos-dashboard-order-public-ip"
+    data-ng-if="$ctrl.step === 1"
```
Comment (antleblanc) [inline packages/manager/modules/anthos/src/dashboard/components/order-public-ip/template.html:3]: **issue:** Not sure that we are supposed to use `<oui-modal>` this way.

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js [hunk 2]
```diff
@@ -1,7 +1,68 @@
...
+  $onInit() {
+    this.$q
+      .all({
+        anthosCatalog: this.AnthosTenantsService.getAnthosCatalog(),
+        expressOrderUrl: this.User.getUrlOf('express_order'),
+      })
+      .then(({ anthosCatalog, expressOrderUrl }) => {
+        this.addon = extractPublicIpsAddonFromAnthosCatalog(anthosCatalog);
+        this.expressOrderUrl = expressOrderUrl;
+        if (!this.addon) throw new Error('missingAddon');
```
Comment (varun257) [inline packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js:29]: Shouldn't this be `displayAlerterMessage` as well?

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js [hunk 3]
```diff
@@ -1,7 +1,68 @@
...
+        planCode: this.addon.planCode,
+        duration: PRICING_DURATION,
+        pricingMode: this.addon.price.mode,
+        quantity: this.quantity,
+        configuration: [],
+      },
+    ];
+
+    window.open(
+      `${this.expressOrderUrl}review?products=${JSURL.stringify(products)}`,
```
Comment (varun257) [inline packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js:63]:
```suggestion
      `${this.expressOrderUrl}review?products=${JSURL.stringify(products)}`, '_blank'
```

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/template.html [hunk 4]
```diff
@@ -1,11 +1,66 @@
...
+    </oui-field>
+</oui-modal>
+<oui-modal
+    class="anthos-dashboard-order-public-ip"
+    data-ng-if="$ctrl.step === 2"
+    data-heading="{{ 'anthos_dashboard_order_public_ip_modal_title' | translate }}"
+    data-primary-action="$ctrl.order()"
+    data-primary-label="{{ 'anthos_dashboard_order_public_ip_modal_action_confirm' | translate }}"
+    data-secondary-action="$ctrl.step = 1"
+    data-secondary-label="{{ 'anthos_dashboard_order_public_ip_modal_action_previous' | translate }}"
```
Comment (varun257) [inline packages/manager/modules/anthos/src/dashboard/components/order-public-ip/template.html:43]: All the labels can use one time binding as well
```suggestion
    data-secondary-label="{{:: 'anthos_dashboard_order_public_ip_modal_action_previous' | translate }}"
```

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js (hunk 2 above)
Comment (y4nnL) [reply packages/manager/modules/anthos/src/dashboard/components/order-public-ip/controller.js:28]: Yes it is in the chained catch handler

File: packages/manager/modules/anthos/src/dashboard/components/order-public-ip/constants.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const ADD_ON_PLAN_CODE = 'option-public-ips-64';
```
Comment (mohammed-zahaf) [inline packages/manager/modules/anthos/src/dashboard/components/order-public-ip/constants.js:1]: I think could be write in one word `ADDON_PLAN_CODE `

Comment (y4nnL) [reply packages/manager/modules/anthos/src/dashboard/components/order-public-ip/constants.js]: :+1:

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**